static/js/message_list_view.js

```diff
@@ -1030,60 +1011,17 @@ export class MessageListView {
             return false;
         }
 
-        const info = $message_viewport.message_viewport_info();
-        const scroll_limit = this._scroll_limit($selected_row, info);
-
-        // This next decision is fairly debatable.  For a big message that
-        // would push the pointer off the screen, we do a partial autoscroll,
-        // which has the following implications:
-        //    a) user sees scrolling (good)
-        //    b) user's pointer stays on screen (good)
-        //    c) scroll amount isn't really tied to size of new messages (bad)
-        //    d) all the bad things about scrolling for users who want messages
-        //       to stay on the screen
-        let scroll_amount;
-        let need_user_to_scroll;
-
-        if (new_messages_height <= scroll_limit) {
-            // This is the happy path where we can just scroll
-            // automatically, and the user will see the new message.
-            scroll_amount = new_messages_height;
-            need_user_to_scroll = false;
-        } else {
-            // Sometimes we don't want to scroll the entire height of
-            // the message, but our callers can give appropriate
-            // warnings if the message is gonna be offscreen.
-            // (Even if we are somewhat constrained here, the message may
-            // still end up being visible, so we do some arithmetic.)
-            scroll_amount = scroll_limit;
-            const offset = $message_viewport.offset_from_bottom($last_visible);
-
-            // For determining whether we need to show the user a "you
-            // need to scroll down" notification, the obvious check
-            // would be `offset > scroll_amount`, and that is indeed
-            // correct with a 1-line message in the compose box.
-            // However, the compose box is open with the content of
-            // the message just sent when this code runs, and
-            // `offset_from_bottom` if an offset from the top of the
-            // compose box, which is about to be reset to empty.  So
-            // to compute the offset at the time the user might see
-            // this notification, we need to adjust by the amount that
-            // the current compose is bigger than the empty, open
-            // compose box.
-            const compose_textarea_default_height = 42;
-            const compose_textarea_current_height = $("#compose-textarea").height();
-            const expected_change =
-                compose_textarea_current_height - compose_textarea_default_height;
-            const expected_offset = offset - expected_change;
-            need_user_to_scroll = expected_offset > scroll_amount;
-        }
-
-        // Ok, we are finally ready to actually scroll.
+        // Find out how close we are to the bottom of the feed. The +1 is to make sure we get
+        // consistent behavior when composing a message right after an autoscroll.
```

Comment:
Can you explain specifically what happens if we don't have this `+1`?

Reply:
Lets say we post a message that triggers the autoscroll. The view would then scroll down to the bottom of the feed. If we then instantly write another message, offset_from_bottom would return 0 since we were just moved to the bottom. The scroll would then not trigger on the second message and our newest message would be hidden from us.

I realize now that a neater solution would be to remove the +1 and use >= on L1017...

Reply:
Yeah, that sounds good -- can you make that tweak?

Reply:
The tweak has been made. However I'm having a slightly different off-by-one problem.

After running system_initiated_animate_scroll with the value from offset_from_bottom, value returned from calling offset_from_bottom again is consistently off by about 0.1 to 1 px.

In order to adress this I changed message_viewport.js's is_scrolled_up method to have an error margin of 1 px.

This seems to work well but I'm not quite sure why the scroll method doesn't actually go all the way down. Do you have any thoughts on this?